app/src/actioncommands.cpp

```diff
@@ -542,6 +543,11 @@ void ActionCommands::duplicateKey()
 {
     Layer* layer = mEditor->layers()->currentLayer();
     if (layer == nullptr) return;
+    if (!layer->visible())
+    {
+        mEditor->getScribbleArea()->showLayerNotVisibleWarning();
```

Comment:
I'd like if we could avoid having actioncommands know scibblearea, instead refer to showLayerNotVisibleWarning through Editor via a proxy method that calls scribblearea.

Reply:
Agreed. I have added a proxy method, please check to see if you think it is an acceptable solution.